toptal/picasso
# fix: make timeline follow table layout

[FX-3801]

### Description

This pull request sets the proper layout of Timeline component, so it keeps the constant width of cells in the same column (table layout)

Design confirmation https://toptal-core.slack.com/archives/CCC3GP6CC/p1698140396833409

### How to test


- [Temploy](https://picasso.toptal.net/fx-3801-fix-timeline-row-s-date-column-to-have-the-same-width-for-all-kind-of-dates)
- All checks should pass
- Compare the timeline in https://staff-portal-pr-11286.toptal.rocks/referrals (https://github.com/toptal/staff-portal/pull/11286) and https://staff-portal.toptal.net/referrals pages
- Check the visual test changes in https://happo.io/a/675/p/1189/compare/7557289dc4e7a2806505560821997b7db136c712/6ef4ceea8ad9479b6d3f6e3fd879fc59585133ce
- Additionally tested in Firefox and Safari

<img width="1095" alt="Screenshot 2023-10-24 at 15 12 09" src="https://github.com/toptal/picasso/assets/1390758/13ecc18d-84fe-4519-8f24-7198591ac2e9">

<img width="1069" alt="Screenshot 2023-10-24 at 15 12 32" src="https://github.com/toptal/picasso/assets/1390758/d2e1e042-722d-48bb-9926-425284ed0564">


### Screenshots

Before

<img width="861" alt="Screenshot 2023-10-23 at 15 36 54" src="https://github.com/toptal/picasso/assets/1390758/8f922849-6f04-4650-9313-4d6842875479">

After

<img width="898" alt="Screenshot 2023-10-24 at 11 10 47" src="https://github.com/toptal/picasso/assets/1390758/bfd6b763-734a-4013-a950-d6090ccbe936">

### Development checks

- [N/A] Create `examples` for component
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-3801]: https://toptal-core.atlassian.net/browse/FX-3801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
